src/login/oidc/redirectHandler/TokenSaver.ts

```diff
@@ -43,43 +42,33 @@ export default class TokenSaver implements ITokenSaver {
   ) {}
 
   async saveTokenAndGetSession(
-    localUserId: string,
+    sessionId: string,
     idToken: string,
     accessToken?: string,
     refreshToken?: string
-  ): Promise<ISessionInfo> {
-    throw new Error("Not Implemented");
-    // const decoded = await this.joseUtility.decodeJWT(
-    //   // TODO this should actually be the id_vc of the token
-    //   accessToken as string
-    // );
-    // // TODO validate decoded token
-    // // TODO extract the localUserId from state and put it in the session
+  ): Promise<void> {
+    // throw new Error("Not Implemented");
+    const decoded = await this.joseUtility.decodeJWT(
+      // TODO this should actually be the id_vc of the token
+      accessToken as string
+    );
+    // TODO validate decoded token
+    // TODO extract the localUserId from state and put it in the session
     // const session = this.sessionCreator.create({
     //   localUserId,
     //   webId: decoded.sub as string,
     //   loggedIn: true
     // });
-    // await this.storageUtility.setForUser(
-    //   session.localUserId,
-    //   "accessToken",
-    //   accessToken as string
-    // );
-    // await this.storageUtility.setForUser(
-    //   session.localUserId,
-    //   "idToken",
-    //   idToken as string
-    // );
-    // await this.storageUtility.setForUser(
-    //   session.localUserId,
-    //   "webId",
-    //   decoded.sub as string
-    // );
-    // await this.storageUtility.setForUser(
-    //   session.localUserId,
-    //   "refreshToken",
-    //   refreshToken as string
-    // );
-    // return session;
+    await this.storageUtility.setForUser(
+      sessionId,
+      {
+        accessToken: accessToken as string,
+        webId: decoded.sub as string,
+        idToken: idToken as string,
+        refreshToken: refreshToken as string,
+        isLoggedIn: "true"
```

Comment:
Are you storing access tokens in session storage? That would be bad

Reply:
Indeed, this is the first issue I want to look at. For some reason, storing them in memory doesn't work with the implicit flow, and anyway a refactor to keep them in a closure is what we really want.

Reply:
Actually no, the `secure: true` part means that the access token is stored in an in-memory store, not the session storage. The bad news is the DPoP token is stored in the session storage currently, so it's not like it's any better.